### PR TITLE
Check instance types are set in get nodegroups

### PR DESCRIPTION
**Pull request by aclevername**
### Description

https://github.com/weaveworks/eksctl/issues/3597
### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/actions/nodegroup/get.go [hunk 1]
```diff
@@ -56,6 +58,17 @@ func (m *Manager) GetAll() ([]*manager.NodeGroupSummary, error) {
 
 	return summaries, nil
 }
+func getInstanceTypes(ng *awseks.Nodegroup) string {
+	if len(ng.InstanceTypes) > 0 {
+		instanceTypes := []string{}
+		for _, instanceType := range ng.InstanceTypes {
+			instanceTypes = append(instanceTypes, *instanceType)
+		}
+		return strings.Join(instanceTypes, ",")
```
Comment (cPu1) [inline pkg/actions/nodegroup/get.go:67]: Minor nit:
```suggestion
		return strings.Join(aws.StringValueSlice(ng.InstanceTypes), ",")
```

**Review by cPu1 — APPROVED** [review]
LGTM. Although it's strange that instance type can be missing from a nodegroup.

**Comment by aclevername** [comment]
> LGTM. Although it's strange that instance type can be missing from a nodegroup.

yeah.. I'm wondering perhaps if the ng is scaled to 0?

**Comment by rlister** [comment]
> LGTM. Although it's strange that instance type can be missing from a nodegroup.

We have some older nodegroups with missing instance type, and ran into this problem with eksctl 0.45. Perhaps an old version of ekctl created nodegroups like this?

**Comment by lloydchang** [comment]
When your time permits, my suggestions are:
1. @karrth @rlister would **redact** any credential and signature from `-v5` output.
2. @karrth @rlister As next steps, share **redacted** `diff` output of `eksctl -v5` — `diff`'ing 0.44 and 0.45.
3. This is in addition to @aclevername's ask to @karrth for steps to reproduce at https://github.com/weaveworks/eksctl/issues/3597#issuecomment-821569888

Thank you @aclevername @cPu1 @Callisto13 @karrth @rlister

-----

There's an `INSTANCE TYPE` of `m5.2xlarge` and node group `MIN SIZE` of `1` in @karrth's post at https://github.com/weaveworks/eksctl/issues/3597#issuecomment-821566509 to @aclevername
```
CLUSTER		NODEGROUP	STATUS
<clustername>	<groupname>	CREATE_COMPLETE
```
```
CREATED
2020-11-19T20:02:34Z
```
```
MIN SIZE	MAX SIZE	DESIRED CAPACITY
1		12		1
```
```
INSTANCE TYPE
m5.2xlarge
```
```
IMAGE ID	ASG NAME
ami-####	eksctl-<clustername>-nodegroup-<groupname>
```

**Comment by aclevername** [comment]
> > LGTM. Although it's strange that instance type can be missing from a nodegroup.
> 
> We have some older nodegroups with missing instance type, and ran into this problem with eksctl 0.45. Perhaps an old version of ekctl created nodegroups like this?
> 
> Creation date: 2020-10-13.

I don't think its related to eksctl, I could be wrong but all we do in `get nodegroups` is present the information returned to us from the EKS api. Its more likely that this field was somehow optional/empty in older managed nodegroups.

**Comment by rlister** [comment]
@lloydchang this is the 44-45 diff:

```
1,4c1,4
< 2021-04-20 11:33:37 [▶]  role ARN for the current session is "arn:aws:sts::<redact>:assumed-role/admin/<redact>"
< 2021-04-20 11:33:37 [ℹ]  eksctl version 0.44.0
< 2021-04-20 11:33:37 [ℹ]  using region us-east-1
< 2021-04-20 11:33:38 [▶]  nodegroups = [{
---
> 2021-04-20 11:33:52 [▶]  role ARN for the current session is "arn:aws:sts::<redact>:assumed-role/admin/<redact>"
> 2021-04-20 11:33:52 [ℹ]  eksctl version 0.45.0
> 2021-04-20 11:33:52 [ℹ]  using region us-east-1
> 2021-04-20 11:33:52 [▶]  nodegroups = [{
116,119d115
< CLUSTER       NODEGROUP       STATUS          CREATED                 MIN SIZE        MAX SIZE        DESIRED CAPACITY    INSTANCE TYPE    IMAGE ID        ASG NAME
< prod1 ng-531bb4ae     CREATE_COMPLETE 2020-10-30T01:49:16Z    4               25              4                           eks-d6babb99-53a8-d412-3e5e-4ce6a744b22e
< prod1 ng-9ea1c657     CREATE_COMPLETE 2021-01-14T19:45:18Z    1               25              1                       c5.4xlarge                   eks-dcbb8137-7b07-05aa-a426-3eac4bcaac85
< prod1 ng-dde21fe5     CREATE_COMPLETE 2021-04-19T20:35:41Z    4               25              4                       m5.2xlarge                   eks-e2bc75ec-af55-6377-1f0c-74aa5d510644
```

I just tested 46-rc.0, and this fixes the problem for our misbehaving nodegroup.

**Comment by lloydchang** [comment]
Thanks @aclevername @rlister 

Related, there is https://github.com/weaveworks/eksctl/pull/3612 which is the node group is scaled to 0 situation.

**Pull request merged**